### PR TITLE
[PC-4423] Share offer improvements

### PR DESCRIPTION
**Pull request by justinemignot**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-4423

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [x] Added new reusable components to the AppComponents page and communicate to the team on slack
- [x] Added a **screenshot** for UI tickets.
- [x] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/offer/services/useShareOffer.ts [hunk 1]
```diff
@@ -11,24 +12,42 @@ import { getLocationName } from '../atoms/LocationCaption'
 
 import { useFunctionOnce } from './useFunctionOnce'
 
-const shareOffer = async (offer: OfferResponse) => {
+const shareOffer = async (offer: OfferResponse, who: string | undefined) => {
   const { id, isDigital, name, venue } = offer
   const locationName = getLocationName(venue, isDigital)
-  const title = _(t`Retrouve ${name} chez ${locationName} sur le pass Culture`)
+  const message = _(t`Retrouve "${name}" chez "${locationName}" sur le pass Culture`)
```
Comment (antoinewg) [inline src/features/offer/services/useShareOffer.ts:18]: traductions à mettre à jour ?

File: src/features/offer/services/useShareOffer.ts [hunk 2]
```diff
@@ -11,24 +12,42 @@ import { getLocationName } from '../atoms/LocationCaption'
...
   const url = `${DEEPLINK_DOMAIN}offer/?id=${id}`
+  const completeMessage = Platform.OS === 'ios' ? message : message.concat(`\n\n${url}`)
+  const title = `${who ? who : 'Je'} t'invite à découvrir une super offre sur le pass Culture !`
```
Comment (antoinewg) [inline src/features/offer/services/useShareOffer.ts:21]:
```suggestion
  const title = `${who ?? 'Je'} t'invite à découvrir une super offre sur le pass Culture !`
```

Comment (antoinewg) [reply src/features/offer/services/useShareOffer.ts:21]: C'est pas bizarre d'avoir son nom dans un message qu'on envoie ? "Antoine t'invite à découvrir" 🤔
J'aurais challenger l'US pour simplifier le code.

File: src/features/offer/services/useShareOffer.ts [hunk 3]
```diff
@@ -11,24 +12,42 @@ import { getLocationName } from '../atoms/LocationCaption'
...
 import { useFunctionOnce } from './useFunctionOnce'
 
-const shareOffer = async (offer: OfferResponse) => {
+const shareOffer = async (offer: OfferResponse, who: string | undefined) => {
   const { id, isDigital, name, venue } = offer
   const locationName = getLocationName(venue, isDigital)
-  const title = _(t`Retrouve ${name} chez ${locationName} sur le pass Culture`)
+  const message = _(t`Retrouve "${name}" chez "${locationName}" sur le pass Culture`)
   const url = `${DEEPLINK_DOMAIN}offer/?id=${id}`
+  const completeMessage = Platform.OS === 'ios' ? message : message.concat(`\n\n${url}`)
```
Comment (antoinewg) [inline src/features/offer/services/useShareOffer.ts:20]: pk `message.concat(`\n\n${url}`)` ? un petit commentaire explicatif serait le bienvenue

Comment (justinemignot) [reply src/features/offer/services/useShareOffer.ts:20]: ok je pensais que les commentaires //ios only //android only suffirait, je modifie

File: src/features/offer/services/useShareOffer.ts (hunk 2 above)
Comment (justinemignot) [reply src/features/offer/services/useShareOffer.ts]: Ok, vu avec Mathieu, bon challenge :)

**Review by antoinewg — APPROVED** [review]

**Pull request merged**